pkg/controller/utils/status.go

```diff
@@ -21,6 +21,9 @@ func SetAccountStatus(awsAccount *awsv1alpha1.Account, message string, ctype aws
 
 // SetAccountClaimStatus sets the condition and state of an accountClaim
 func SetAccountClaimStatus(awsAccountClaim *awsv1alpha1.AccountClaim, message string, reason string, ctype awsv1alpha1.AccountClaimConditionType, state awsv1alpha1.ClaimStatus) {
+	if awsAccountClaim == nil {
+		return
+	}
```

Comment:
@iamkirkbater this shouldn't be required or we should move the error handling here instead of adding this validation in two places.

Reply:
Everywhere this is called (once in account_controller.go, thrice in byoc.go), it is immediately followed by a `r.Client.Status().Update(context.TODO(), accountClaim)`. IMO it is worth refactoring to
- include that `Update` in this method;
- include this `nil` check; and
- return an `error` value to the caller.

Reply:
We have a card open to refactor this directly into the AccountClaim struct itself: https://issues.redhat.com/browse/OSD-7194.

So this nil check will only persist until that is refactored, as if we can't get the accountClaim then there's no struct to call the update on, and we'll have to handle that in each of those calls respectively.

Reply:
At a 5-minute perusal, that refactor doesn't seem like it would be a whole lot bigger than this change. Why wait? :)

("This NPE is causing regular problems in the field" is a valid answer btw.)

Reply:
"This NPE is causing regular problems in the field" ;)

Also, just from my personal experience "oh, this refactor should be really simple and quick" rarely happens to turn out that way hahaha.  At least with our current test suite :/  So I'd rather punt that and tackle it the right way then try to rush a refactor as a fix.